[#9510] Instructor: Course Page/Student List Component: Implement sorting for table headers

pbrahmbhatt3 opened the pull request:
Fixes #9510 

Added sort functions in instructor-courses-page.components.ts to sort activeCourses, archivedCourses and softDeletedCourses. 
Added buttons on in instructor-courses-page.component.html to sort the course list.

I would like to know more about how would you like to sort the studentList component as it is already sorted by team name.

pbrahmbhatt3 commented:
> 
> 
> Hi @pbrahmbhatt3, these parts of your pull request do not appear to follow our [contributing guidelines](https://github.com/TEAMMATES/teammates/blob/master/docs/process.md#step-4-submit-a-pr):
> 
>     1. PR Title
>        
>        * Issue Reference (`#<issue-number>`) missing.

I have updated PR's title to include issue number

pbrahmbhatt3 commented:
@tanhengyeow  I have updated the files correctly to fix the issue. However, I am not sure why the test is failing. Could please advice me in what could I correct to solve it! Your help will be much appreciated

pbrahmbhatt3 commented:
I found out that there is an error in feedback-questions.service.ts. 
![image](https://user-images.githubusercontent.com/27025266/55271150-b6ef3e00-527e-11e9-907e-fd04aa0e94f6.png)

It has nothing to do with the changes I have made. I am not sure how to fix it

RonakLakhotia commented on package.json:
```diff
@@ -48,9 +48,9 @@
   "devDependencies": {
     "@angular-builders/jest": "~7.1.2",
-    "@angular-devkit/build-angular": "~0.11.4",
+    "@angular-devkit/build-angular": "^0.13.7",
```
Revert this unrelated change

RonakLakhotia replied:
Also, don't forget to revert these changes

RonakLakhotia commented on package.json:
```diff
@@ -48,9 +48,9 @@
+    "@angular-devkit/build-angular": "^0.13.7",
     "@angular/cli": "~7.1.4",
-    "@angular/compiler-cli": "~7.1.4",
+    "@angular/compiler-cli": "^7.2.11",
```
Revert this too

RonakLakhotia commented on src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.html:
```diff
@@ -121,6 +131,16 @@ <h2>Active Courses</h2>
+      <div class="float-right">
+        <h5 class="d-inline"><strong> Sort By: </strong></h5>
+        <div class="btn-group" data-toggle="buttons">
+          <button class="btn btn-light" (click)="sortById('archieve')">Course ID</button>
```
Not sure what you mean by archieve here, do you mean archive?

RonakLakhotia commented on src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts:
```diff
@@ -82,10 +82,10 @@ export class InstructorCoursesPageComponent implements OnInit {
-              private statusMessageService: StatusMessageService,
-              private courseService: CourseService,
-              private modalService: NgbModal) { }
+    private httpRequestService: HttpRequestService,
```
Why this change in indentation?

RonakLakhotia commented on src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts:
```diff
@@ -208,7 +208,7 @@ export class InstructorCoursesPageComponent implements OnInit {
         this.statusMessageService.showErrorMessage(resp.error.message);
       });
-    }, () => {});
+    }, () => { });
```
Unrelated change

RonakLakhotia commented on src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts:
```diff
@@ -270,7 +305,7 @@ export class InstructorCoursesPageComponent implements OnInit {
       });
 
-    }, () => {});
+    }, () => { });
```
Unrelated change

pbrahmbhatt3 commented:
Hi @RonakLakhotia,

Thank you for your suggestions! I have updated my code. I have followed Instructor-home-page component to implement sorting for instructor-courses-page. I added 2 fields called SortBy and instructorCourseSortBy to implement sorting. 

However, the tests fail saying  Received value does not match stored snapshot "InstructorCoursesPageComponent should snap with default fields 1". Do I also have to update the snapshot of this component to include these fields?

RonakLakhotia commented on src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts:
```diff
@@ -223,6 +223,84 @@ export class InstructorCoursesPageComponent implements OnInit {
+  sortPanelsBy(by: string):
+      ((a: ActiveCourse, b: ActiveCourse) => number) {
+    return ((a: ActiveCourse, b: ActiveCourse): number => {
+      let strA: string;
```
These variables name don't seem to be intuitive. Perhaps use a more relatable name?

xpdavid commented:
Close due to inactivity.
